freetextresponse/freetextresponse.py

```diff
@@ -496,14 +485,3 @@ def submit(self, data, suffix=''):
             'submitted_message': self._get_submitted_message(),
         }
         return result
-
-
-class Credit(Enum):
```

Comment:
While not strictly required here, I'm generally a fan of enums and constants for values like this. The main benefits are:
- easier to read
  - it's immediately obvious what the effect of `Credit.full` vs. `Credit.half` is, without needing to figure out what the scale is
- easier to update
  - just change the values once, instead of everywhere they occur
     - right now the code is able to cast those integers `1` and `0` to `float`, but what if the grading functionality is changed to require `float`s to be passed in? Then we're back to tracking down every single occurrence; this is error prone.
- easier to trace/search
  - if I want to track down all credit-related code paths, I could (prior to this) search for `Credit`; if we change it to literal values, you risk finding many false positives, as there could be plenty of lines that contain the number `0`.

I think you may have overthought this a bit;
I can see two simpler, less invasive approaches:
- just cast from enum to float where needed (ie, the call to grade.publish)
- turn those enums into `float` constants
  - this gives us the same benefits; single definition, human-readable name, etc, but allows us to maintain the intended data type (`float`)

```python
CREDIT_FULL = 1.0
CREDIT_HALF = 0.5
CREDIT_NONE = 0.0
```